### PR TITLE
Add a lexer for dune files in doc

### PR DESCRIPTION
**Pull request by emillon**
This replaces the uses of "scheme" and "lisp" lexers.

The lexer is fairly non-opinionated and tries to stick to just the lexing of dune-lang (atoms, strings, things like pforms, etc), and does not have a list of known stanzas and field names for example.

It does two special things:

- it recognizes metasyntax like `<arg>` and highlights it accordingly to show that these parts are not meant to be put verbatim in dune files
- it highlights the first atom in lists in a different way. This corresponds to how most of dune-lang works, but is not totally correct either. so that part can be made optional.

### CODE REVIEW

**Comment by emillon** [comment]
Example:

lexer

**Comment by Alizter** [comment]
Looks good so far. I wonder if we can color the `<fields>` so that they are also orange like the monospace we have int eh rest of the doc.

**Comment by emillon** [comment]
> I wonder if we can color the `<fields>` so that they are also orange like the monospace we have int eh rest of the doc.

Not fan of putting orange everywhere, but I can suggest something a bit different: we can refer to metasyntax using italics in the text so that they have the same look, like this. I pushed an example.

> With this option, a single parser named *<base_name>* is generated.

2 questions:
- should we do the highlighting of first atoms? this can be deactivated by commenting line 28 of the lexer. my opinion is that we're highlighting almost nothing when it's deactivated and it's giving the right results in almost all cases, so that should be kept.
- should I add a "lexical conventions" section somewhere? that would be to clarify that parts that look like *\<that>* are not meant to be added verbatim.

**Comment by Alizter** [comment]
> * should we do the highlighting of first atoms? this can be deactivated by commenting line 28 of the lexer. my opinion is that we're highlighting almost nothing when it's deactivated and it's giving the right results in almost all cases, so that should be kept.

Yes, there are very few cases where we don't want the highlighting (if at all).

> * should I add a "lexical conventions" section somewhere? that would be to clarify that parts that look like _<that>_ are not meant to be added verbatim.

I would suggest a section in hacking.rst although perhaps @rgrinberg has a better suggestion.

File: doc/stanzas/menhir.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ is:
 
 - ``(merge_into <base_name>)`` is used to define modular parsers. This
   correspond to the ``--base`` command line option of ``menhir``. With this
-  option, a single parser named ``base_name`` is generated.
+  option, a single parser named *<base_name>* is generated.
```
Comment (Alizter) [inline doc/stanzas/menhir.rst:32]: This seems like a sensible thing to do generally.

**Review by christinerose — APPROVED** [review]

**Pull request merged**